### ISSUE TITLE
AI v2 React Plugin only work with react router?

### ISSUE THREAD

**Issue by andrewconnell** [microsoft/ApplicationInsights-JS#922]
I have a React app that will not use the router as the UX doesn't work well with how the router works (using [Office UI Fabric's Pivot control](https://developer.microsoft.com/en-us/fabric#/controls/web/pivot)). I've set up my app following the sample as best I can see, and I see page views getting logged form my app, but nothing about component usage stats.

I set up a **TelemetryService**...

```ts
import { ApplicationInsights } from "@microsoft/applicationinsights-web";
import { ReactPlugin } from "@microsoft/applicationinsights-react-js";

class TelemetryService {
  public reactPlugin: ReactPlugin;
  private _appInsights: ApplicationInsights;

  constructor() {
    this.reactPlugin = new ReactPlugin();
  }

  public initialize(reactPluginConfig) {
    let INSTRUMENTATION_KEY = '*********-****-****-****-***********';

    this._appInsights = new ApplicationInsights({
      config: {
        instrumentationKey: INSTRUMENTATION_KEY,
        maxBatchInterval: 0,
        disableFetchTracking: false,
        extensions: [this.reactPlugin],
        extensionConfig: {
          [this.reactPlugin.identifier]: reactPluginConfig
        }
      }
    });
    this._appInsights.loadAppInsights();
  }
}

export let telemetryService = new TelemetryService();
```

... and initialize it within the host for my app (a SharePoint web part):

```ts
export default class NewsletterBuilderWebPart extends BaseClientSideWebPart<INewsletterBuilderWebPartProps> {

  public onInit(): Promise<void> {
    return new Promise<void>((resolve, reject) => {
      const history = createBrowserHistory({ basename: '' });
      telemetryService.initialize({ history: history });
      resolve();
    });
  }

  public render(): void {
    const element = React.createElement(RootContainer, {});
    ReactDom.render(element, this.domElement);
  }

  protected onDispose(): void {
    ReactDom.unmountComponentAtNode(this.domElement);
  }
..
}
```

... and I've added the `withAITracking` to the `<RootContainer />` component as well as all the other components rendered in the app listed in the code below... 

```tsx
export class RootContainer extends React.Component<IRootContainerProps, {}> {
  public render(): React.ReactElement<IRootContainerProps> {
    return (
      <div className={ styles.newsletterBuilder }>
        <Label style={{ fontSize: FontSizes.size42 }}>Voitanos Newsletter Builder Utility</Label>
        <Pivot>
          <PivotItem headerText="Dashboard">
            <DashboardContainer />
          </PivotItem>
          <PivotItem headerText="DeDupe Scrubber">
            <DedupeContainer />
          </PivotItem>
          <PivotItem headerText="Categorizer">
            <CategorizerContainer />
          </PivotItem>
          <PivotItem headerText="Item Processor">
            <ItemProcessorContainer />
          </PivotItem>
          <PivotItem headerText="Sorter">
            <ItemSorterContainer />
          </PivotItem>
        </Pivot>
        <GutterContainer />
      </div>
    );
  }
}

export default withAITracking(telemetryService.reactPlugin, RootContainer);
```

But when watching the network traffic in the browser, i only see a single `track` call with a pageview event... nothing related to React.

Am I _holding it wrong_ or does the plugin only work with the React router?

**Comment by markwolff** [comment]
I can't tell from your example, but the export statement for each of your components must also be wrapped with `withAITracking` to get component usage tracking
https://github.com/microsoft/ApplicationInsights-JS/tree/master/vNext/extensions/applicationinsights-react-js#basic-usage

**Comment by andrewconnell** [comment]
**TL;DR**
You weren't seeing all the code... but your response helped put 1+1 together on how the plugin works after looking at it's source. I've got it working now.

**More...**
If you look at the last snippet in my OP, notice two export statements. I'm using barrels to export components. When I was doing this, the componet that was being used was the class I defined, _not_ the return from the `withAITracking` function.

So... I had to go back into my components, remove `export` from their definitions, and update their barrels to export them by name... so change this:

```ts
export * from './RootContainer';
export * from './IRootContainerProps';
```

to this...

```ts
export {default as RootContainer} from './RootContainer';
export * from './IRootContainerProps';
```

Now I see the events... Thanks!
